### ISSUE TITLE
Time dependent collapse operators (str-format) in mcsolve don't work

### ISSUE THREAD

**Issue by TheIdealis** [qutip/qutip#844]
If I use a time-dependent collapse operator in `mcsolve`:

``` python
import numpy as np
from qutip import destroy, basis, mcsolve

b = destroy(5)
H = b.dag() * b
collapse = [[b, '9 * exp(-(t / 5.) ** 2)']]
psi0 = basis(5, 1)

times = np.linspace(0, 100, 100)
data = mcsolve(H, psi0, times, collapse, [b.dag() * b]) 
```

I get the following error:

``` python
Error compiling Cython file:
------------------------------------------------------------
...
@cython.boundscheck(False)
@cython.wraparound(False)
def col_expect(int which, double t, complex[::1] data, int[::1] idx, int[::1] ptr, complex[::1] vec):

    if which == 0:
        out *= conj(9 * exp(-(t / 5.) ** 2))
       ^
------------------------------------------------------------
```

I looked into `cy/codegen.py` and found, that the line

``` python
out_string.append("    out *= conj(" + self.c_tdterms[ind] + ")")
```

of the function `func_which_expect(self)` (currently line 355) should be replaced with

``` python
out_string.append("    out = conj(" + self.c_tdterms[ind] + ")")
```

**Comment by nathanshammah** [comment]
This seems an easy fix to mcsolve, on it, thank you.

**Comment by nathanshammah** [comment]
Now solved by @Ericgig in release 4.4.0.
